modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/saml/SAMLECPSSOTestCase.java

```diff
@@ -0,0 +1,223 @@
+            Assert.assertTrue(result.contains("urn:oasis:names:tc:SAML:2.0:status:Success"),"Successfully Authenticated the user for "+config);
+            Assert.assertTrue(result.contains("<SOAP-ENV:Envelope xmlns:SOAP-ENV=\"http://schemas.xmlsoap.org/soap/envelope/\">"),"Successfully retrieved the SAML Response bound with SOAP");
+
+        } catch (Exception e) {
+            Assert.fail("SAML ECP Login test failed for " + config, e);
+        }
+    }
+
+    @Test(description = "Testing SAML ECP login failure" ,groups = "wso2.is" ,dependsOnMethods = {"testSAMLECPLogin"})
+    public void testSAMLECPAuthnFailLogin(){
+        try {
+            httpClient = new DefaultHttpClient();
+            HttpResponse response;
+            String samlECPReq = buildECPSAMLRequest(SAML_ECP_ACS_URL ,SAML_ECP_ISSUER);
+            response = Utils.sendECPPostRequest(SAML_ECP_SSO_URL,USER_AGENT, httpClient, config.getUser().getUsername() ,"RandomPassword", samlECPReq );
+            String result = extractDataFromResponse(response);
+            log.debug("This is the Result "+result);
+            int responseCode = response.getStatusLine().getStatusCode();
+            log.debug(responseCode);
+            Assert.assertEquals(responseCode, 200, "Successful login failure response returned code " + responseCode);
+            Assert.assertTrue(result.contains("urn:oasis:names:tc:SAML:2.0:status:AuthnFailed"),"Successfully  identified the login failure "+config);
+        } catch (Exception e) {
+            Assert.fail("SAML ECP Login failure test failed for " + config, e);
+        }
+    }
+
+    @Test(description = "Testing SAML ECP SOAP faults for invalid requests" ,groups = "wso2.is" ,dependsOnMethods = {"testSAMLECPAuthnFailLogin"})
+    public void testSOAPFault(){
+        try {
+            httpClient = new DefaultHttpClient();
+            HttpResponse response;
+            String samlECPReq =buildInvalidECPSAMLRequest(SAML_ECP_ACS_URL ,SAML_ECP_ISSUER);
+            response = Utils.sendECPPostRequest(SAML_ECP_SSO_URL,USER_AGENT, httpClient, config.getUser().getUsername() ,config.getUser().getPassword(), samlECPReq );
+            String result = extractDataFromResponse(response);
+            log.debug("This is the Result "+result);
+            int responseCode = response.getStatusLine().getStatusCode();
+            log.debug(responseCode);
+            Assert.assertEquals(responseCode, 500, "Successfully returned a SOAP fault with internal server error " + responseCode);
+        } catch (Exception e) {
+            Assert.fail("SAML ECP SOAP fault test failed for " + config, e);
+        }
+    }
+
+    @Test(description = "Remove service provider", groups = "wso2.is", dependsOnMethods = { "testSOAPFault" })
+    public void testRemoveSP()
+            throws Exception {
+        Boolean isAddSuccess = ssoConfigServiceClient.removeServiceProvider(config.getApp().getArtifact());
```

Comment:
isAddSuccess does not imply that you are trying to removeservice provider.
Shall we use a variable like, 'isServiceProviderRemoved' or something which implies the SP removal is happening?

Reply:
> isServiceProviderRemoved

updated the variable name with 'isServiceProviderRemoved'